src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -416,7 +417,8 @@ protected void CheckDocumentIsOpen(DocumentId documentId)
         // TODO: switch this protected once we have confidence in API shape
         internal void OnSourceGeneratedDocumentOpened(
             SourceGeneratedDocumentIdentity documentIdentity,
-            SourceTextContainer textContainer)
+            SourceTextContainer textContainer,
+            SourceGeneratedDocument document)
```

Comment:
Passing along the document here seems simple/reasonable. I'm not sure if there's other surprises here we might run into. We still have pending work for VS for Mac to also wire some of this up which I hope will result in a bit of refactoring of stuff out of the VS layer which might also make this a bit saner too.